hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -278,6 +279,32 @@ func (r *HostedClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		return ctrl.Result{}, fmt.Errorf("failed to get cluster %q: %w", req.NamespacedName, err)
 	}
 
+	res, err := r.reconcile(ctx, req, log, hcluster)
+	condition := metav1.Condition{
+		Type:               string(hyperv1.ReconciliationSucceeded),
+		ObservedGeneration: hcluster.Generation,
+		Status:             metav1.ConditionTrue,
+		Message:            "Reconciliaton Succeeded",
+	}
+	if err != nil {
+		condition.Status = metav1.ConditionFalse
+		condition.Reason = "ReconciliationError"
+		condition.Message = err.Error()
+	}
+	old := meta.FindStatusCondition(hcluster.Status.Conditions, string(condition.Type))
```

Comment:
condition.Type? did you mean `string(hyperv1.ReconciliationSucceeded)`?

Reply:
It ends up being the same thing but I agree that is more explicit, updated